lib/immutable-struct.rb

```diff
@@ -36,26 +36,47 @@ class ImmutableStruct
   #     p.minor    # => "yup"
   #     p.minor?   # => true
   #
-  def self.new(*attributes,&block)
+  # === Derived attributes
+  # A derived attribute is a zero-arity method (ie. getter) that applies a transformation to the
+  # existing values stored inside an +ImmutableStruct+. For instance:
+  #
+  #    ImmutableStruct.new(:foo, :bar) do
+  #      def derived; self.foo + ":" + self.bar; end
+  #    end
+  #
+  # Please note that, unless overriden, +ImmutableStruct#to_h+ will serialize all zero-arity methods.
+  #
+  #
+  def self.new(*attributes, &block)
+    raise ArgumentError if attributes && Array(attributes).empty?
+
     klass = Class.new do
+      array_attr = lambda { |attr| attr.kind_of?(Array) && attr.size == 1 }
+      boolean_attr = lambda { |attr| String(attr).match(/(^.*)\?$/) }
+
       attributes.each do |attribute|
-        if attribute.to_s =~ /(^.*)\?$/
+        case attribute
+          when ''
```

Comment:
This is just a defensive hotfix until I fix and report the issue to the Rubinius and JRuby projects. IMHO, it's good that the users of those implementations can reliably use this gem.

If this line is deleted, Rubinius would just silently create an empty `ImmutableStruct` and JRuby would blow up with a naked Java exception. Additionally, this test should be commented-out:

`expect { ImmutableStruct.new('') }.to raise_error(NameError)`

Again, maybe I'm getting ahead of myself here and it'd be fine if I just delete the `when` and comment-out or skip the test.